tests/unit/backend/recipe/algorithm/test_PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -0,0 +1,241 @@
+        instrumentDefinitionFile = "/SNS/SNAP/shared/Calibration/Powder/SNAPLite.xml"
+        groupingFile = "/SNS/SNAP/shared/Calibration/Powder/PixelGroupingDefinitions/SNAPFocGroup_Column.lite.nxs"
+        referenceParametersFile = "/SNS/SNAP/shared/Calibration/Powder/04bd2c53f6bf6754/Column_lite_parameters.json"
+
+        with pytest.raises(RuntimeError) as excinfo:
+            run_test(
+                instrumentDefinitionFile=instrumentDefinitionFile,
+                calibrationState="junk",
+                groupingFile=groupingFile,
+                referenceParametersFile=referenceParametersFile,
+                reverseGroupingIndex=True,
+            )
+        assert "Calibration" in str(excinfo.value)
+
+    def run_test(
+        instrumentDefinitionFile,
+        calibrationState,
+        groupingFile,
+        referenceParametersFile,
+        reverseGroupingIndex,
+    ):
+        """Test execution of PixelGroupingParametersCalculationAlgorithm"""
+        pixelGroupingAlgo = PixelGroupingParametersCalculationAlgorithm()
+        pixelGroupingAlgo.initialize()
+
+        pixelGroupingAlgo.setProperty("InputState", calibrationState)
+        pixelGroupingAlgo.setProperty("InstrumentDefinitionFile", instrumentDefinitionFile)
+        pixelGroupingAlgo.setProperty("GroupingFile", groupingFile)
+
+        assert pixelGroupingAlgo.execute()
+
+        # parse the algorithm output and create a list of PixelGroupingParameters
+        pixelGroupingParams_str = json.loads(pixelGroupingAlgo.getProperty("OutputParameters").value)
+        pixelGroupingParams_calc = []
+        for item in pixelGroupingParams_str:
+            pixelGroupingParams_calc.append(PixelGroupingParameters.parse_raw(item))
+
+        # parse the reference file. Note, in the reference file each kind of parameter is grouped into its own list
+        f = open(referenceParametersFile)
+        pixelGroupingParams_ref = json.load(f)
+
+        # compare calculated and reference parameters
+        number_of_groupings_calc = len(pixelGroupingParams_calc)
+        assert len(pixelGroupingParams_ref["twoTheta"]) == number_of_groupings_calc
+        assert len(pixelGroupingParams_ref["dMin"]) == number_of_groupings_calc
+        assert len(pixelGroupingParams_ref["dMax"]) == number_of_groupings_calc
+        assert len(pixelGroupingParams_ref["delDOverD"]) == number_of_groupings_calc
+
+        # reverseGroupingIndex takes care of the different order of pixel groupings between "full" anf "lite"
+        # instruments. This has to do with how Mantid treats different kinds of grouping files used by
+        # PixelGroupingParametersCalculationAlgorithm
+        index = 0 if reverseGroupingIndex else number_of_groupings_calc - 1
+        for param in pixelGroupingParams_ref["twoTheta"]:
+            assert abs(float(param) - pixelGroupingParams_calc[index].twoTheta) < 1.0e-5
+            index += 1 if reverseGroupingIndex else -1
```

Comment:
`index += (1 if reverseGroupingIndex else -1)` is easier to understand at a glance

Reply:
Done.